src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Roslyn4.0.cs

```diff
@@ -19,7 +20,11 @@ public partial class LoggerMessageGenerator : IIncrementalGenerator
         public void Initialize(IncrementalGeneratorInitializationContext context)
         {
             IncrementalValuesProvider<ClassDeclarationSyntax> classDeclarations = context.SyntaxProvider
-                .CreateSyntaxProvider(static (s, _) => Parser.IsSyntaxTargetForGeneration(s), static (ctx, _) => Parser.GetSemanticTargetForGeneration(ctx))
+                .ForAttributeWithMetadataName(
+                    context,
+                    Parser.LoggerMessageAttribute,
+                    (node, _) => node is MethodDeclarationSyntax,
+                    (context, _) => context.TargetNode.Parent as ClassDeclarationSyntax)
```

Comment:
the original code did no actual semantic checks *beyond* just checking that there was an attribute with a particular fully qualified name on it.  this is unlike the regex generator which does additional semantic checks.

Reply:
I believe that was intentional.  https://github.com/dotnet/runtime/pull/51064/files#r614207808

Reply:
Maybe @chsienki and @jaredpar can advise about it?

Reply:
note: this wasn't a complaint.  Just pointing out the difference between these two generators.  It's not necessarily right/wrong, just something to potentially look into depending on what semantics you want.